llvm/lib/Transforms/Utils/ScalarEvolutionExpander.cpp

```diff
@@ -43,6 +43,37 @@ cl::opt<unsigned> llvm::SCEVCheapExpansionBudget(
 
 using namespace PatternMatch;
 
+PoisonFlags::PoisonFlags(const Instruction *I) {
+  NUW = false;
+  NSW = false;
+  Exact = false;
+  Disjoint = false;
+  NNeg = false;
+  if (auto *OBO = dyn_cast<OverflowingBinaryOperator>(I)) {
+    NUW = OBO->hasNoUnsignedWrap();
+    NSW = OBO->hasNoUnsignedWrap();
```

Comment:
Shouldn't this be `hasNoSignedWrap`? Same as below when setting.

Reply:
Indeed! Thanks for catching it.